edgedb/protocol/codecs/namedtuple.pyx

```diff
@@ -114,3 +115,27 @@ cdef class NamedTupleCodec(BaseNamedRecordCodec):
         codec.fields_codecs = fields_codecs
 
         return codec
+
+    def _make_missing_args_error_message(self, args):
+        required_args = set()
+
+        for i in range(len(self.fields_codecs)):
+            name = datatypes.record_desc_pointer_name(self.descriptor, i)
+            required_args.add(name)
+
+        passed_args = set(args.keys())
+        missed_args = required_args - passed_args
+        extra_args = passed_args - required_args
+
+        error_message = f'expected {required_args} keyword arguments'
+        error_message += f', got {passed_args}'
+
+        missed_args = set(required_args) - set(passed_args)
+        if missed_args:
+            error_message += f', missed {missed_args}'
+
+        extra_args = set(passed_args) - set(required_args)
+        if extra_args:
+            error_message += f', extra {extra_args}'
+
+        return RuntimeError(error_message)
```

Comment:
Let's change the error type from `RuntimeError` to `edgedb.QueryArgumentError`.

Reply:
Changed. Also squashed commits.